tests/test_xopen.py

```diff
@@ -566,3 +566,67 @@ def test_xopen_zst_fails_when_zstandard_not_available(monkeypatch):
     with pytest.raises(ImportError):
         with xopen.xopen(TEST_DIR / "file.txt.zst", mode="rb", threads=0) as f:
             f.read()
+
+
+# tests for passing in filehandles
+def test_passing_in_gz_filehandles_for_reading():
+    first_line = CONTENT_LINES[0].encode("utf-8")
+    with open(TEST_DIR / "file.txt.gz", "rb") as fh:
+        with xopen(fh, "rb") as f:
+            assert f.readline() == first_line
+
+
+def test_passing_in_xz_filehandles_for_reading():
```

Comment:
Since this test is a copy of the one above with only the file extension changed, you can merge them into one. Just copy it how it is done above in other tests that use the `@pytest.mark.parametrize("ext", extensions)` decorator. This ensures that you test all the extensions that are supported (`.zst` is only supported optionally), and it also includes the empty extension (no compression), for which a test is missing at the moment.